sycl/include/CL/__spirv/spirv_types.hpp

```diff
@@ -129,10 +136,12 @@ enum class MatrixLayout : uint32_t {
 // information to SPIRV translator.
 // The long term solution would be to introduce a matrix type in Clang and use
 // it instead of this member.
-template <typename T, std::size_t R, std::size_t C, MatrixLayout U,
+template <typename T, std::size_t R, std::size_t C, MatrixLayout L, MatrixUse U,
           Scope::Flag S = Scope::Flag::Subgroup>
 struct __spirv_JointMatrixINTEL {
-  T (*Value)[R][C][static_cast<size_t>(U) + 1][static_cast<size_t>(S) + 1];
+  T(*Value)
+  [R][C][static_cast<size_t>(L) + 1][static_cast<size_t>(U) + 1]
```

Comment:
Use will be the last and optional parameter in SPIR-V Type declaration instruction. Could we also move it to be last dimension in the array (not as template parameter)?